src/Cli/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -55,7 +55,7 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
-                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    if (_workingDirectory is not null && _workingDirectory.Length != 0)
```

Comment:
`string.IsNullOrEmpty` does not have the proper attributes in .NET Framework to be recognized as a null-check. Therefore, these cause errors since we build for both TFMs. There are several other options to avoid this problem, but are all generally workarounds. On the day I removed all `!` uses (called the [null-forgiving operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving)), I also converted all the `string.IsNullOrEmpty` to have the functional equivalent check like this (for the projects being converted in this PR).

Reply:
Have you considered only running nullable checks when targeting core? Highly recommend in multi targeted projects.

Reply:
I have but I'm not doing this as a project. This PR was something I started as a side-thing in March and only revived the PR as there was renewed interest in making the repo nullability compatible. If someone on my team wants to make a decision on the direction of this, then that's fine. All I've done it make an old PR buildable and express the problems around nullability to @marcpopMSFT.

You've suggested some completely reasonable ways to handle this stuff but I don't even know if this is a project we want to get done yet or what the timeframe would be. Only "decision" we made was we didn't want to use the null-forgiving operator (!) when it can be avoided. This repo doesn't have strong, identifiable standards for coding as it is an amalgamation of many different repos.